config/routes.rb

```diff
@@ -65,9 +65,9 @@
   match '/settings/profile', to: 'user#update', via: 'patch', as: :update_user_profile
   match '/settings/profile_info', to: 'user#update_profile', via: 'patch', as: :update_user_profile_info
 
-  match '/settings/theme', to: 'user#edit_theme', via: 'get', as: :edit_user_theme
-  match '/settings/theme', to: 'user#update_theme', via: 'patch', as: :update_user_theme
-  match '/settings/theme/delete', to: 'user#delete_theme', via: 'delete', as: :delete_user_theme
+  match '/settings/theme', to: 'settings/theme#edit', via: 'get', as: :edit_user_theme
+  match '/settings/theme', to: 'settings/theme#update', via: 'patch', as: :update_user_theme
+  match '/settings/theme/delete', to: 'settings/theme#destroy', via: 'delete', as: :delete_user_theme
```

Comment:
might want to take the opportunity to make the route definitions a bit less noisy, e.g.

```suggestion
  namespace :settings do
    get :theme, to: redirect('/settings/theme/edit')
    resource :theme, controller: :theme, only: %i[edit update delete]
  end
  resolve('Theme') { [:settings_theme] } # to make link_to/form_for work nicely when passing a `Theme` object to it, see also: https://api.rubyonrails.org/v6.1.5.1/classes/ActionDispatch/Routing/Mapper/CustomUrls.html#method-i-resolve
```

this will generate the routes like this (you'll then need to update all the referenced url helpers + the hardcoded routes in the typescript files):

```
             Prefix Verb   URI Pattern                    Controller#Action
     settings_theme GET    /settings/theme(.:format)      redirect(301, /settings/theme/edit)
edit_settings_theme GET    /settings/theme/edit(.:format) settings/theme#edit
     settings_theme PATCH  /settings/theme(.:format)      settings/theme#update
                    PUT    /settings/theme(.:format)      settings/theme#update
```